pkg/controller/cni/cilium_reconciler.go

```diff
@@ -0,0 +1,245 @@
+package cni
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/controllers/remote"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	eksacluster "github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	"github.com/aws/eks-anywhere/pkg/controller"
+	"github.com/aws/eks-anywhere/pkg/controller/serverside"
+	"github.com/aws/eks-anywhere/pkg/executables"
+	"github.com/aws/eks-anywhere/pkg/networking/cilium"
+)
+
+const defaultRequeueTime = time.Second * 10
+
+type CiliumReconciler struct {
+	Client  client.Client
+	Log     logr.Logger
+	tracker *remote.ClusterCacheTracker
+}
+
+func (cr *CiliumReconciler) Reconcile(ctx context.Context, cluster *anywherev1.Cluster, capiCluster *clusterv1.Cluster, specWithBundles *eksacluster.Spec) (controller.Result, error) {
```

Comment:
Instead of passing the capi cluster, it might be more flexible if we pass a `client.Client` directly. That way this reconciler becomes agnostic from the fact that it will be used with a remote client (provided by the tracker.

And we can implement the client generation one level higher.

Reply:
done